lib/bedrock.js

```diff
@@ -320,6 +243,55 @@ api.exit = function() {
   cluster.worker.kill();
 };
 
+async function _runOnce({type, id, fn, options}) {
+  // notify master to schedule work (if not already scheduled/run)
+  process.send({type, id, options});
+
+  // wait for scheduling result
+  let msg = await _waitForOneMessage({type, id});
```

Comment:
should this come before process.send to ensure that the local listener is registered?

Reply:
I guess this is what you were saying about why you did it the way it was before.  I guess this is fine now, is that right?

Reply:
This is what I was saying before -- but because of "run to completion" rules it shouldn't matter. If it did matter, I'd have to put it back to how it was before -- being called as `promise = _waitForOneMessage()` before `process.send` and then `await promise` after.